File: 백준/Silver/10157. 자리배정/자리배정.java


import java.util.Scanner;

public class Main {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		int c = sc.nextInt(); // 이게 j
		int r = sc.nextInt(); // 이게 i
		int k = sc.nextInt();

		int[] row = { -1, 0, 1, 0 };
		int[] col = { 0, 1, 0, -1 };

		int cnt = 1;
		int i = r - 1;
		int j = 0;
		int d = 0;
		int[][] grid = new int[r][c];
		grid[i][j] = cnt++;
		while (cnt <= r * c) {
			int nr = i + row[d];
			int nc = j + col[d];
			if (nr < 0 || nr >= r || nc < 0 || nc >= c || grid[nr][nc] != 0) {
				d = (d + 1) % 4;
				continue;
			}
			grid[nr][nc] = cnt++;
			i = nr;
			j = nc;

		}
		int x = 0;
		int y = 0;
		boolean bool = false;
		for (int ii = 0; ii < r; ii++) {
			for (int jj = 0; jj < c; jj++) {
				if (grid[ii][jj] == k) {
					bool = true;
					x = jj+1;
					y =r - ii;
				} 
			}
		}
		if(bool) {
			System.out.println(x+" "+y);
		}
		else {
			System.out.println(0);
		}
	}

}

File: 백준/Silver/10157. 자리배정/자리배정.py
c, r = map(int, input().split())
num = int(input())

grid = [[0] * c for i in range(r)]
# for row in grid:
#     print(*row)

ridx = r - 1
cidx = 0
row = [-1, 0, 1, 0]
col = [0, 1, 0, -1]
d = 0
n = 1
grid[ridx][cidx] = n
ansX = -1
ansY = -1
while n < r * c:

    n += 1
    if (0 > ridx + row[d] or ridx + row[d] >= r or 0 > cidx + col[d] or cidx + col[d] >= c or grid[ridx + row[d]][
        cidx + col[d]] != 0):
        d = (d + 1) % 4
    ridx = ridx + row[d]
    cidx = cidx + col[d]
    grid[ridx][cidx] = n
    if(n==num):
        ansX = r-ridx
        ansY = cidx+1
# for row in grid:
#     print(*row)
if(ansX == -1 and ansY == -1):
    if(num!=1):
        print(0)
    else:
        print("1 1")
else:
    print(ansY, ansX)
